### ISSUE TITLE
Replace <?php echo '...' ?> with <?= '...' ?>

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#3277]
### Issue / Bug  / Question / New Feature

This cleans the code up significantly in the views especially.  IMO we should do it in a future minor release (after CI 4.1.3 upgrade)

**Comment by ebentil** [comment]
I wouldn't advise that. It won't work on php instances that have `short_open_tag` disabled

**Comment by objecttothis** [comment]
@ebentil Thank you for pointing this out.  Short open tags `<?` are being deprecated by PHP starting with 7.4 and removed entirely by 9.0 (https://wiki.php.net/rfc/deprecate_php_short_tags_v2).  The short open tag is different from echo shortcut tag (`<?=`) (see the article here https://leehblue.com/summary-php-echo-shortcut-tag/ and the section titled "Backward Incompatible Changes" in the first link).

The TL;DR is that the echo shortcut tag is not affected by the status of `short_open_tag` which will be deprecated anyway since the tag conflicts with xml.  I'm not saying this means we are making the change, but so far, I haven't seen any cons other than the work it would take to refactor the code.

**Comment by daN4cat** [comment]
What's CI4 style recommendation on the matter?
I would suggest to be aligned with their code style suggestions to avoid having personal views on the matter.

**Comment by objecttothis** [comment]
CI4 does not yet a style guide (that I could find).  There is one for CI3 which states that short open tags `<? ... ?>` should never be used due to what @ebentil pointed out but that the echo shortcut tag `<?= $foo ?>` would always be available regardless of the value of `short_open_tag` since PHP 5.4.

Ryan Seachrest's style guide says that it should be used when possible. https://gist.github.com/ryansechrest/8138375#5-short-echo-tag

Older sources seems to say to avoid it for one of two reasons: 1- short-open-tags being possibly disabled (which now has no effect on the echo shortcut tag, or 2- stated opinion without backup as to why.

PSR-1 only states that either <?php or <?= should be used, but doesn't say either way whether <?php echo ... is better or <?= ... is better.

PSR-2 and PSR-12 don't address it.

PHP just says it's an option and that the semicolon is not needed if using the short echo shortcut.  https://www.php.net/manual/en/language.basic-syntax.phptags.php

Verdict:
- Seems like either is OK syntactically, neither is encouraged more by official style guides and it just comes down to preference, but maybe I'm wrong in my understanding.

If I'm right, then we just need to decide if the added clarity of `<?= $foobar ?>` over `<?php echo $foobar; ?>` is worth the refactoring and devs getting used to the standard.

I'm OK with either, but prefer the shorthand. If people feel strongly, then I'm OK with leaving it.

**Comment by objecttothis** [comment]
I posted the question in the CI Best Practices forum (https://forum.codeigniter.com/thread-80167.html).

Here is CakePHP framework's stance on it: https://book.cakephp.org/3/en/contributing/cakephp-coding-conventions.html#short-echo

**Comment by objecttothis** [comment]
One user on the thread pointed out that CI examples use short echo tag

https://codeigniter4.github.io/CodeIgniter4/outgoing/views.html#creating-loops (see last example).
